notebooks/source/truncated_distributions.ipynb

```diff
@@ -0,0 +1,1916 @@
+{
```

Comment:
<p>It is typically to get constant samples for <code>low</code> if its density is too large comparing to other values. For this model, I believe the posterior has a high density at low=4. However, the setting is a bit unusual to me. It is lucky that the data seems to have minimum is 4, hence prior for <code>low</code> is <code>Categorical(0.2 * jnp.ones(5))</code> make sense. If the data have minimum is 3, we should set a different prior. I think you can use the model</p><pre class="ql-syntax" spellcheck="false">def truncated_poisson_model(num_observations, x=None, infer_low=True):
    if infer_low:
  &nbsp; &nbsp;   low = numpyro.sample("low", dist.Categorical(0.2 * jnp.ones((5,))))
    else:
        low = 4
  &nbsp; rate = numpyro.sample("rate", dist.LogNormal(1, 1))
  &nbsp; with numpyro.plate("observations", num_observations):
  &nbsp; &nbsp; &nbsp; numpyro.sample("x", LeftTruncatedPoisson(rate, low), obs=x)
</pre><p>, then draw prior samples with <code>infer_low=False</code>. Or you can use the prior</p><pre class="ql-syntax" spellcheck="false">dist.Categorical(logits=jnp.zeros(int(x.min() + 1) if x is not None else 5)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyro-ppl/numpyro/pull/1272/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>